cli/landscape.py

```diff
@@ -13,46 +13,126 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import os
+import enum
 from util import ctx
 from util import (
     info,
     which,
     warning,
     CliHints,
     CliHint,
+    existing_dir,
 )
 import landscape_setup.concourse as setup_concourse
-import landscape_setup.whd as setup_whd
 import landscape_setup.monitoring as setup_monitoring
+import landscape_setup.secrets_server as setup_secrets_server
+import landscape_setup.whd as setup_whd
 
-
-def deploy_or_upgrade_concourse(
-    config_name: CliHint(typehint=str, help="the cfg_set to use"),
-    deployment_name: CliHint(typehint=str, help="namespace and deployment name")='concourse',
+class LandscapeComponent(enum.Enum):
+    CONCOURSE='concourse'
+    SECRETS_SERVER='secrets server'
+    WHD='webhook dispatcher'
+    MONITORING='monitoring'
+
+def deploy_or_upgrade_landscape(
+    config_set_name: CliHint(typehint=str, help="the name of the config set to use"),
+    components: CliHint(typehint=[str], help=(
+        f"list of components to deploy. Possible Values: "
+        f"{', '.join([c.value for c in LandscapeComponent])}. By default, ALL components will "
+        f"be deployed.")
+    )=None,
+    webhook_dispatcher_chart_dir: CliHint(
+        typehint=str,
+        help="directory of webhook dispatcher chart",
+    )=None,
+    concourse_deployment_name: CliHint(
+        typehint=str, help="namespace and deployment name for Concourse"
+    )='concourse',
     timeout_seconds: CliHint(typehint=int, help="how long to wait for concourse startup")=180,
+    webhook_dispatcher_deployment_name: str='webhook-dispatcher',
     dry_run: bool=True,
 ):
-    '''Deploys a new concourse-instance using the given deployment name and config-directory.'''
-    which("helm")
+    '''Deploys the given components of the Concourse landscape.
+    '''
+    known_components = [component.value for component in LandscapeComponent]
+    # handle default (all known components)
+    if not components:
```

Comment:
if changed as suggest above, components will either be the empty list, or a non-empty list of `LandscapeComponent`. If empty, add all enum values. No need to convert to str, is there?